pyqtgraph/graphicsItems/tests/test_PlotDataItem.py

```diff
@@ -114,14 +114,36 @@ def test_clipping():
     w.addItem(c)
 
     c.setClipToView(True)
-    w.setXRange(200, 600)
-    for x_min in range(100, 2**10 - 100, 100):
-        w.setXRange(x_min, x_min + 100)
-
+    for x_min in range(-200, 2**10 - 100, 100):
+        x_max = x_min + 100
+        w.setXRange(x_min, x_max, padding=0)
         xDisp, _ = c.getData()
-        vr = c.viewRect()
```

Comment:
This never updates without redraw, and is delayed even if processEvents calls are added.